### PR TITLE
S10 t3 integrate the front end and back end for the book appointment functionality

### PR DESCRIPTION
**Pull request by bradleyhallett**
1. Calls the back end api for workers given a schoolId on the select worker page. 
2. Calls api for avalibles for a given worker (specified on previous page) on book appointment page
3. Calls book appointment api to record an appointment booking after student fills in all information required on book Appointment page.
4. Error handling for the responses on both select worker and book appointment page. 
5. Unit tests for SelectWorker.js, SelectWorkerForm.js, BookAppointment.js and BookAppointmentForm.js

6. Changes button colours from red to more appropriate colours on log in, sign up, select worker and book appointment page. 

- [ ]  Run through the process of booking an appointment for a student and ensure the new appointment is displayed on the users home page. 

**I realized an issue with the worker availability logic on the book appointment page, the times presented to the student on this page can coincide with already booked appointments with other workers. Adjusting the worker availability api must be planned for in the next sprint.   

**please note:** run the website in **chrome** as Safari was showing issues loading the final page of the appointment booking process. This is unfortunately out of scope for this sprint.

### CODE REVIEW

**Comment by Simran230** [comment]
Hi Bradley! Great job done! I see that you have 100% coverage on your SelectWorker.js test. Where would the reminder 3 of the tests be? I'm sorry perhaps I cannot see it properly. 
image

Thank you very much! 
Simran

**Comment by bradleyhallett** [comment]
> Hi Bradley! Great job done! I see that you have 100% coverage on your SelectWorker.js test. Where would the reminder 3 of the tests be? I'm sorry perhaps I cannot see it properly.
> image
> 
> Thank you very much!
> Simran

Hi Simran, 

My other test files are CreateAppointment.js, CreateAppointmentForm.js and SelectWorkeForm.js. Unfortunately, despite my best efforts I was not able to achieve 100% coverage on all my files being added, however I did manage to create test cases for each file.

**Comment by Simran230** [comment]
Hello Bradley, 

Fantastic job on setting up the front end and changing the colour of the UI. While testing the code I noticed just a small spelling mistake in the word "available". Perhaps we can say, "Sorry there are no available times for this worker"? 

image

Thanks a lot! 

Simran

**Comment by Simran230** [comment]
> > Hi Bradley! Great job done! I see that you have 100% coverage on your SelectWorker.js test. Where would the reminder 3 of the tests be? I'm sorry perhaps I cannot see it properly.
> > image
> > Thank you very much!
> > Simran
> 
> Hi Simran,
> 
> My other test files are CreateAppointment.js, CreateAppointmentForm.js and SelectWorkeForm.js. Unfortunately, despite my best efforts I was not able to achieve 100% coverage on all my files being added, however I did manage to create test cases for each file.

Hi Bradley, 

Thanks for informing me of this! My apologies for not being able to tell about the rest of your tests! 

Thank you! 

Simran

**Comment by bradleyhallett** [comment]
> Hello Bradley,
> 
> Fantastic job on setting up the front end and changing the colour of the UI. While testing the code I noticed just a small spelling mistake in the word "available". Perhaps we can say, "Sorry there are no available times for this worker"?
> 
> image
> 
> Thanks a lot!
> 
> Simran

Hi Simran, 

I have fixed the spelling mistake you mentioned above. Thank you for pointing this out!

-Bradley

**Comment by Simran230** [comment]
> > Hello Bradley,
> > Fantastic job on setting up the front end and changing the colour of the UI. While testing the code I noticed just a small spelling mistake in the word "available". Perhaps we can say, "Sorry there are no available times for this worker"?
> > image
> > Thanks a lot!
> > Simran
> 
> Hi Simran,
> 
> I have fixed the spelling mistake you mentioned above. Thank you for pointing this out!
> 
> -Bradley

Thank you Bradley! I appreciate it! 

Simran

File: react-ui/src/components/Common/ColorSchema.scss [hunk 1]
```diff
@@ -13,3 +13,5 @@ $emergency-btn-color: rgb(255, 175, 25);
 
 $typical-font-size: 20px;
 $input-font-size: 16px;
+
+$login-signup-button-colour: #b8b1fd;
```
Comment (sehersarin) [inline react-ui/src/components/Common/ColorSchema.scss:17]: I see you have changed the button colours for the login and signup pages to be consistent with the header colour - as per TA reviews. This looks good - thank you!

File: react-ui/src/components/CreateAppointmentForm.test.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
+import React from "react";
```
Comment (sehersarin) [inline react-ui/src/components/CreateAppointmentForm.test.js:1]: From our conversation, I understand that low file coverage is expected for the CreateAppointmentForm.test.js and SelectWorkerForm.test.js tests however, both CreateAppointment.test.js and SelectWorker.test.js files have 100% test coverage. Remaining tests will be in scope for another sprint.

**Comment by sehersarin** [comment]
Hi Bradley, to keep consistent with the UI of the rest of the application can you please add a header to this page. Also, there is a type if you could kindly change "lefts" to "left".

Screen Shot 2020-12-02 at 4 44 11 PM

Additionally, even after entering a reason I am not able to click on the submit button and get this error pop-up. Is this expected? 

Screen Shot 2020-12-02 at 4 46 07 PM

File: react-ui/src/components/SelectWorkerForm.test.js [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+    const { container } = render(<BrowserRouter>
+      < SelectWorkerForm user={studentUser}/>
+    </BrowserRouter>);
+    await waitFor(() => expect(axios.post).toHaveBeenCalled())
+
+    // Assert
+    expect(container.innerHTML.includes("Joshua Brooks")).toBe(true);
+  });
+/*
+  test("Testing case where user does not select a worker", async () => {
```
Comment (sehersarin) [inline react-ui/src/components/SelectWorkerForm.test.js:124]: I understand this commented code has been left in the PR as this test case needs to be further debugged (out of scope for this ticket) and some of the existing code here can be reused.

File: react-ui/src/components/Common/ColorSchema.scss (hunk 1 above)
Comment (bradleyhallett) [reply react-ui/src/components/Common/ColorSchema.scss:17]: Thank you for noticing!

File: react-ui/src/components/CreateAppointmentForm.test.js (hunk 2 above)
Comment (bradleyhallett) [reply react-ui/src/components/CreateAppointmentForm.test.js:1]: Yes! The remaining coverage of the CreateAppointmentForm.js and SelectWorkerForm.js will be in scope for another sprint. Thank you for asking!

File: react-ui/src/components/SelectWorkerForm.test.js (hunk 3 above)
Comment (bradleyhallett) [reply react-ui/src/components/SelectWorkerForm.test.js:124]: Yes, the commented code in this test file may be used for further testing in the future, however it was not working as expected and so I decided to leave it for future development.

**Comment by bradleyhallett** [comment]
> Hi Bradley, to keep consistent with the UI of the rest of the application can you please add a header to this page.
> 
> Screen Shot 2020-12-02 at 4 44 11 PM
> 
> Additionally, even after entering a reason I am not able to click on the submit button and get this error pop-up. Is this expected?
> 
> Screen Shot 2020-12-02 at 4 46 07 PM

Very interesting, I do not have this issue on my end, the header does in fact populate on this page and I am able to submit with a filled in reason and a time selected. Did you call npm run start:dev when booting up the application?

**Comment by sehersarin** [comment]
> > Hi Bradley, to keep consistent with the UI of the rest of the application can you please add a header to this page.
> > Screen Shot 2020-12-02 at 4 44 11 PM
> > Additionally, even after entering a reason I am not able to click on the submit button and get this error pop-up. Is this expected?
> > Screen Shot 2020-12-02 at 4 46 07 PM
> 
> Very interesting, I do not have this issue on my end, the header does in fact populate on this page and I am able to submit with a filled in reason and a time selected. Did you call npm run start:dev when booting up the application?

Hi Bradley, thank you for fixing the typo! This functionality glitch is a browser error as it doesn't work correctly on Safari however, works fine on Google Chrome. Please make a comment regarding this above and for now, it can be in scope for another ticket.

**Comment by bradleyhallett** [comment]
> > > Hi Bradley, to keep consistent with the UI of the rest of the application can you please add a header to this page.
> > > Screen Shot 2020-12-02 at 4 44 11 PM
> > > Additionally, even after entering a reason I am not able to click on the submit button and get this error pop-up. Is this expected?
> > > Screen Shot 2020-12-02 at 4 46 07 PM
> > 
> > 
> > Very interesting, I do not have this issue on my end, the header does in fact populate on this page and I am able to submit with a filled in reason and a time selected. Did you call npm run start:dev when booting up the application?
> 
> Hi Bradley, thank you for fixing the typo! This functionality glitch is a browser error as it doesn't work correctly on Safari however, works fine on Google Chrome. Please make a comment regarding this above and for now, it can be in scope for another ticket.

Thank you for identifying this issue, I believe you are correct in saying this issue is out of scope and it will be planned for in another ticket. I have made a comment above in the PR description.

**Comment by mtatran** [comment]
Hi Bradley, this looks great. However, in Chrome it does not connect to the back-end. I am not able to see any timeslots or reason for booking.

**Comment by bradleyhallett** [comment]
> Hi Bradley, this looks great. However, in Chrome it does not connect to the back-end. I am not able to see any timeslots or reason for booking.

Hi @mtatran , 

For some reason I am not able to view the picture you took, could you try sending it again?

-Bradley

**Comment by mtatran** [comment]
> > Hi Bradley, this looks great. However, in Chrome it does not connect to the back-end. I am not able to see any timeslots or reason for booking.
> 
> Hi @mtatran ,
> 
> For some reason I am not able to view the picture you took, could you try sending it again?
> 
> -Bradley

Hi Bradley, 

I just checked your latest commit. This is working perfectly on Chrome. However, tests are not passing, please update the snapshots.

**Review by mtatran — CHANGES_REQUESTED** [review]
update snapshots for tests to pass

**Comment by bradleyhallett** [comment]
> > > Hi Bradley, this looks great. However, in Chrome it does not connect to the back-end. I am not able to see any timeslots or reason for booking.
> > 
> > 
> > Hi @mtatran ,
> > For some reason I am not able to view the picture you took, could you try sending it again?
> > -Bradley
> 
> Hi Bradley,
> 
> I just checked your latest commit. This is working perfectly on Chrome. However, tests are not passing, please update the snapshots.

Hi Melissa, 

I have updated the snapshot test to all pass!

-Bradley

**Comment by mtatran** [comment]
> > > > Hi Bradley, this looks great. However, in Chrome it does not connect to the back-end. I am not able to see any timeslots or reason for booking.
> > > 
> > > 
> > > Hi @mtatran ,
> > > For some reason I am not able to view the picture you took, could you try sending it again?
> > > -Bradley
> > 
> > 
> > Hi Bradley,
> > I just checked your latest commit. This is working perfectly on Chrome. However, tests are not passing, please update the snapshots.
> 
> Hi Melissa,
> 
> I have updated the snapshot test to all pass!
> 
> -Bradley

It looks good. I will approve.

**Review by mtatran — APPROVED** [review]

**Pull request merged**